onnxruntime/core/graph/graph.cc

```diff
@@ -2904,6 +2905,8 @@ Status Graph::ReplaceInitializedTensor(const ONNX_NAMESPACE::TensorProto& new_in
     return true;
   };
 
+  ORT_RETURN_IF_NOT(!is_external || utils::HasExternalData(old_initializer), "Trying to replace non-external initializer with external data");
```

Comment:
nit: Double negatives are difficult to read.

Reply:
I could not come up with a better code in this case.

Reply:
ORT_RETURN_IF(is_external && !utils::HasExternalData(old_initializer), ...)